query/executor_test.go

```diff
@@ -595,6 +597,116 @@ func TestQueryExecutor_InvalidSource(t *testing.T) {
 	}
 }
 
+func TestQueryExecutor_WriteQueryToLog(t *testing.T) {
+	q, err := influxql.ParseQuery(`SELECT count(value) FROM cpu`)
+	require.NoError(t, err, "parse query")
+
+	f, err := os.CreateTemp("", "query-test.log")
+	require.NoError(t, err, "create temp file")
+
+	defer os.Remove(f.Name())
+
+	e := NewQueryExecutor()
+	e.WithLogWriter(e.Logger, f.Name())
+
+	e.StatementExecutor = &StatementExecutor{
+		ExecuteStatementFn: func(stmt influxql.Statement, ctx *query.ExecutionContext) error {
+			require.Equal(t, uint64(1), ctx.QueryID, "query ID")
+			return nil
+		},
+	}
+
+	discardOutput(e.ExecuteQuery(q, query.ExecutionOptions{}, nil))
+	err = f.Close()
+	require.NoError(t, err, "close temp file")
+
+	dat, err := os.ReadFile(f.Name())
+	cont := strings.Contains(string(dat), "SELECT count(value) FROM cpu")
+	require.True(t, cont, "expected query output")
+}
+
+// Test to ensure that watcher creates new file on file rename
+func TestQueryExecutor_WriteQueryToLog_WatcherRemoveFile(t *testing.T) {
```

Comment:
It appears that CI doesn't like these tests where on my local machine they pass no problem. I may need to adjust how I'm doing this 🤔